odoo/orm/fields_properties.py

```diff
@@ -105,23 +105,27 @@ def convert_to_cache(self, value, record, validate=True):
         if not value:
             return None
 
-        if isinstance(value, dict):
-            # avoid accidental side effects from shared mutable data
-            return copy.deepcopy(value)
-
-        if isinstance(value, str):
-            value = json.loads(value)
-            if not isinstance(value, dict):
-                raise ValueError(f"Wrong property value {value!r}")
-            return value
+        if not isinstance(value, (str, dict, list)):
+            raise ValueError(f"Wrong property type {type(value)!r}")
 
         if isinstance(value, list):
             # Convert the list with all definitions into a simple dict
             # {name: value} to store the strict minimum on the child
             self._remove_display_name(value)
-            return self._list_to_dict(value)
+            value = self._list_to_dict(value)
+        elif isinstance(value, str):
+            value = json.loads(value)
+            if not isinstance(value, dict):
+                raise ValueError(f"Wrong property value {value!r}")
+        if not validate or not record._ids:
+            return value
+        # writing the same truthy value to properties with different definitions doesn't make sense
+        definition = self._get_properties_definition(record[0])
```

Comment:
but it's not correct, can we check that the definition record is the same and raise if not ?

Reply:
I don't think we can any use case like that. But I agree it is better to raise error instead of keeping silent.
I add the check in the `_get_properties_definition`